gr/lessons/specifics/ecto.md

```diff
@@ -0,0 +1,340 @@
+---
+layout: page
+title: Ecto
+category: specifics
+order: 2
+lang: gr
+---
+
+Το Ecto είναι ένα επίσημο Elixir project το οποίο παρέχει ένα κάλυμμα βάσης δεδομένων και μια ενσωματωμένη γλώσσα ερωτημάτων.  Με το Ecto είμαστε σε θέση να δημιουργούμε μετατροπές, να ορίζουμε μοντέλα, να εισάγουμε και επεξεργαζόμαστε εγγραφές και να τις εξετάζουμε.
+
+{% include toc.html %}
+
+## Εγκατάσταση
+
+Για να ξεκινήσουμε πρέπει να συμπεριλάβουμε το Ecto και έναν αντάπτορα βάσης δεδομένο στο `mix.exs` του project μας.  Μπορείτε να βρείτε μια λίστα υποστηριζόμενων ανταπτόρων στον τομέα [Usage](https://github.com/elixir-lang/ecto/blob/master/README.md#usage) του Ecto README.  Για το παράδειγμά μας θα χρησιμοποιήσυμε την PostgreSQL:
+
+```elixir
+defp deps do
+  [{:ecto, "~> 1.0"},
+   {:postgrex, ">= 0.0.0"}]
+end
+```
+
+Τώρα μπορούμε να προσθέσουμε το Ecto και τον αντάπτορά μας στη λίστα εφαρμογών:
+
+```elixir
+def application do
+  [applications: [:ecto, :postgrex]]
+end
+```
+
+### Αποθετήριο
+
+Τέλος χρειάζεται να δημιουργήσουμε το αποθετήριο του project μας, το κάλυμμα της βάσης δεδομένων.  Αυτό μπορεί να γίνει μέσω της εργασίς `mix ecto.gen.repo`.  Θα καλύψουμε τις εργασίες mix του Ecto στη συνέχεια.  Το αποθετήριο μπορεί να βρεθεί στο `lib/<όνομα του project>/repo.ex`:
+
+```elixir
+defmodule ExampleApp.Repo do
+  use Ecto.Repo,
+    otp_app: :example_app
+end
+```
+
+### Επιτηρητής
+
+Εφόσον έχουμε δημιουργήσει το αποθετήριό μας χρειαζόμαστε να ρυθμίσουμε το δέντρο επιτήρησής μας, το οποίο συνήθως βρίσκεται στο `lib/<όνομα του project>.ex`.
+
+Είναι σημαντικό να σημειώσουμε ότι ορίζουμε το αποθετήριο σαν επιτηρητή με την `supervisor/3` και _όχι_ με την `worker/3`.  Αν δημιουργήσετε την εφαρμογή σας με την σημαία `--sup`, αρκετά από αυτά υπάρχουν ήδη:
+
+```elixir
+defmodule ExampleApp.App do
+  use Application
+
+  def start(_type, _args) do
+    import Supervisor.Spec
+
+    children = [
+      supervisor(ExampleApp.Repo, [])
+    ]
+
+    opts = [strategy: :one_for_one, name: ExampleApp.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
+```
+
+Για περισσότερες πληροφορίες στους επιτηρητές ελέγξτε το μάθημα [Επιτηρητές OTP](/lessons/advanced/otp-supervisors).
```

Comment:
I think that this link should lead to the Greek lesson, not the English one.
`[Επιτηρητές OTP](/lessons/advanced/otp-supervisors)` → `[Επιτηρητές OTP](/gr/lessons/advanced/otp-supervisors)`.